modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/files/ShpFiles.java

```diff
@@ -838,6 +841,15 @@ public StorageFile getStorageFile(ShpFileType type) throws IOException {
 
     public String getTypeName() {
         String path = SHP.toBase(urls.get(SHP));
+        try {
```

Comment:
Hmm... the what if a slash is url encoded, because it's part of the name instead of being part of the path?
What about first determining the name component, and only after that, url decoding it?